### ISSUE TITLE
When using the email verification process user is inserted first

### ISSUE THREAD

**Issue by jccompagnon** [coolcsn/CsnUser#10]
When using the email verification the user is Inserted in the database then the email is sent. 
If the email sent errors out the user is created in the database and needs to be manualy deleted.
The user should not be inserted in the DB till the email is successfully sent

**Comment by coolcsn** [comment]
Yes, you are right. We have to switch the order of the operations. You can send me a PR (Pull Request) and I will merge it, if you like. Or I will add it as an issue and it will be fixed.
